spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -83,7 +83,11 @@ public Long endSnapshotId() {
   }
 
   public String branch() {
-    return confParser.stringConf().option(SparkReadOptions.BRANCH).parseOptional();
+    return confParser
+        .stringConf()
+        .sessionConf(SparkReadOptions.BRANCH)
```

Comment:
I don't think this should be the Spark read option. Session configs are defined in `SparkSQLProperties` and use a form like `spark.sql.iceberg._____`. In this case, how about `spark.sql.iceberg.read-branch`?

I think we also need to be careful about the behavior of this, and possibly use a different `SparkReadConf` option for it. When the SQL read branch is set, we want to use that if it is present and otherwise fall back to the main branch. That may make this complicated enough that we should leave the SQL session read branch out of this commit and add it in a follow-up.

Reply:
@rdblue @amogh-jahagirdar  i am going to remove all session related changes from this PR so that can raise a follow up one. We can support only write by spark option in this PR.

Reply:
@namrathamyske, what about adding back the write side? I think the problem is just with the read side and we can't test the write side of this PR without the SQL option.

Reply:
@rdblue if we just have write side and not read side how can we use session conf ? i think spark sql operations uses scans to read as i explained in https://github.com/apache/iceberg/pull/6651#discussion_r1098126130.